akd/src/directory.rs

```diff
@@ -50,17 +50,30 @@ impl AkdLabel {
 pub struct Directory<S, V> {
     storage: S,
     vrf: V,
+    read_only: bool,
+    /// The cache lock guarantees that the cache is not
+    /// flushed mid-proof generation. We allow multiple proof generations
+    /// to occur (RwLock.read() operations can have multiple) but we want
+    /// to make sure no generations are underway when a cache flush occurs
+    /// (in this case we do utilize the write() lock which can only occur 1
+    /// at a time and gates further read() locks being acquired during write()).
+    cache_lock: Arc<tokio::sync::RwLock<()>>,
 }
 
 impl<S: Storage + Sync + Send, V: AkdVRF> Directory<S, V> {
     /// Creates a new (stateless) instance of a auditable key directory.
     /// Takes as input a pointer to the storage being used for this instance.
     /// The state is stored in the storage.
-    pub async fn new<H: Hasher>(storage: &S, vrf: &V) -> Result<Self, AkdError> {
-        if Directory::<S, V>::get_azks_from_storage(storage)
-            .await
-            .is_err()
-        {
+    pub async fn new<H: Hasher>(storage: &S, vrf: &V, read_only: bool) -> Result<Self, AkdError> {
+        let azks = Directory::<S, V>::get_azks_from_storage(storage, false).await;
+
+        if read_only && azks.is_err() {
+            return Err(AkdError::Directory(DirectoryError::Storage(
+                StorageError::GetData(
+                    "AZKS record not found and Directory constructed in read-only mode".to_string(),
+                ),
+            )));
+        } else if azks.is_err() {
```

Comment:
What do you think on expecting not_found error here instead of any error? I wonder could azks exist but get_azks_from_storage() having transient error?

Reply:
The caller should handle all transient error logic if they want to